src/Features/Core/Portable/ExtractMethod/ExtractMethodResult.cs

```diff
@@ -57,18 +51,17 @@ internal class ExtractMethodResult
 
         internal ExtractMethodResult(
             OperationStatusFlag status,
-            IEnumerable<string> reasons,
+            ImmutableArray<string> reasons,
             Document? documentWithoutFinalFormatting,
             ImmutableArray<AbstractFormattingRule> formattingRules,
             SyntaxToken invocationNameToken,
             SyntaxNode? methodDeclarationNode)
         {
             Status = status;
 
-            Succeeded = status.Succeeded() && !status.HasSuggestion();
-            SucceededWithSuggestion = status.Succeeded() && status.HasSuggestion();
+            Succeeded = status.Succeeded();
 
-            Reasons = (reasons ?? SpecializedCollections.EmptyEnumerable<string>()).ToReadOnlyCollection();
+            Reasons = reasons.NullToEmpty();
```

Comment:
>NullToEmpty [](http://example.com/codeflow?start=30&length=11)

I understand what this is doing, but this name confuses me.

Reply:
this was in case someone was passing 'null' along originally.  With the prior code, you'd get an empty enumerable.  the same now holds true for the immutablearray version.